src/overlays/actors/ovl_Obj_Tsubo/z_obj_tsubo.c

```diff
@@ -46,39 +73,275 @@ static ColliderCylinderInit D_80BA1B94 = {
+    pos.z = this->actor.world.pos.z;
+    floorY = BgCheck_EntityRaycastFloor4(&globalCtx->colCtx, &floorPoly, &bgID, &this->actor, &pos);
+    if (floorY > BGCHECK_Y_MIN) {
+        this->actor.world.pos.y = floorY;
+        Math_Vec3f_Copy(&this->actor.home.pos, &this->actor.world.pos);
+        return true;
+    } else {
+        osSyncPrintf("地面に付着失敗\n");
+        return false;
+    }
+}
+
+void ObjTsubo_InitCollider(Actor* thisx, GlobalContext* globalCtx) {
+    ObjTsubo* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &D_80BA1B94);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+}
+
+void ObjTsubo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjTsubo* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ObjTsubo_InitCollider(&this->actor, globalCtx);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    if (ObjTsubo_SnapToFloor(this, globalCtx) == 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->objTsuboBankIndex = Object_GetIndex(&globalCtx->objectCtx, *(sObjectIds + (((this->actor.params >> 8) & 1))));
+    if (this->objTsuboBankIndex < 0) {
+        osSyncPrintf("Error : バンク危険！ (arg_data 0x%04x)(%s %d)\n", this->actor.params, "../z_obj_tsubo.c", 0x19A);
+        Actor_Kill(&this->actor);
+    } else {
+        func_80BA152C(this);
+        osSyncPrintf("(dungeon keep 壷)(arg_data 0x%04x)\n", this->actor.params);
+    }
+}
+
+void ObjTsubo_Destroy(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    ObjTsubo* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void ObjTsubo_AirBreak(ObjTsubo* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 rand;
+    s16 angle;
+    Vec3f pos;
+    Vec3f velocity;
+    f32 sins;
+    f32 coss;
+    s32 arg5;
+    s32 i;
+
+    for (i = 0, angle = 0; i < 15; i++, angle += 0x4E20) {
+        sins = Math_SinS(angle);
+        coss = Math_CosS(angle);
+
+        pos.x = sins * 8.0f;
+        pos.y = (Rand_ZeroOne() * 5.0f) + 2.0f;
+        pos.z = coss * 8.0f;
+
+        velocity.x = pos.x * 0.23f;
+        velocity.y = (Rand_ZeroOne() * 5.0f) + 2.0f;
+        velocity.z = pos.z * 0.23f;
+
+        Math_Vec3f_Sum(&pos, &this->actor.world.pos, &pos);
+
+        rand = Rand_ZeroOne();
+        if (rand < 0.2f) {
+            arg5 = 96;
+        } else if (rand < 0.6f) {
+            arg5 = 64;
+        } else {
+            arg5 = 32;
+        }
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, &this->actor.world.pos, -240, arg5, 10, 10, 0,
+                             (Rand_ZeroOne() * 95.0f) + 15.0f, 0, 32, 60, KAKERA_COLOR_NONE,
+                             sObjectIds[(this->actor.params >> 8) & 1], D_80BA1B8C[(this->actor.params >> 8) & 1]);
+    }
+
+    func_80033480(globalCtx, &this->actor.world.pos, 30.0f, 4, 0x14, 0x32, 1);
```

Comment:
those last two numbers can be decimal
this is a dust spawner and those numbers are scale and scale step